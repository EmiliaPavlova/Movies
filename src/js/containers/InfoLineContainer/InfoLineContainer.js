import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import HomeResults from '../../components/HomeResults/HomeResults';
import DetailsResults from '../../components/DetailsResults/DetailsResults';

const content = {
  home: 'HomeResults',
  details: 'DetailsResults'
};

const page = 'details';


console.log(content[page]);

class InfoLineContainer extends Component {
  render() {
    return (
      <Fragment>
        {page === 'home'
          ? <HomeResults />
          : <DetailsResults />}
      </Fragment>
    )
  }
}

InfoLineContainer.propTypes = {
  page: PropTypes.oneOf(['home', 'details'])
};

export default InfoLineContainer;