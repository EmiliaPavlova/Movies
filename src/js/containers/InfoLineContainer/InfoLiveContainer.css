.infoLine {
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 40px;
  padding-left: 80px;
  padding-right: 80px;
  font-weight: bold;
  font-size: 12px;
  background: #F5F5F5;
}