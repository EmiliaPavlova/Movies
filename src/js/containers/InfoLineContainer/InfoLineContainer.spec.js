import React from 'react';
import { mount } from 'enzyme';
import InfoLineContainer from './InfoLineContainer';
import './InfoLineContainer.css';

const componentProps = (overrides = {}) => ({
  page: 'home',
  moviesFound: 2,
  sortBy: 'release_date',
  ...overrides
});

const mountInfoLineContainer = props => mount(<InfoLineContainer {...props} />);

describe('InfoLineContainer', () => {
  describe('Snapshot', () => {
    it('matches the snapshot for Home page', () => {
      const wrapper = mountInfoLineContainer(componentProps());
      expect(wrapper).toMatchSnapshot('Home page');
    });
    it('matches the snapshot for Details page', () => {
      const wrapper = mountInfoLineContainer(componentProps({page: 'details'}));
      expect(wrapper).toMatchSnapshot('Details page');
    })
  });
  describe('Home page', () => {
    it('Should return InfoLine', () => {
      const props = componentProps();
      const wrapper = mountInfoLineContainer(props);
      console.log(wrapper.debug());
    })
  })
});
