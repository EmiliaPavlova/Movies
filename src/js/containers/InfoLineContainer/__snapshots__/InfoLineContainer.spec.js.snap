// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoLineContainer Snapshot matches the snapshot 1`] = `
<InfoLineContainer>
  <div
    className="infoLine"
  >
    <DetailsResults>
      <div>
        Films by 
        Drama
         genre
      </div>
    </DetailsResults>
  </div>
</InfoLineContainer>
`;

exports[`InfoLineContainer Snapshot matches the snapshot for Details page: Details page 1`] = `
<InfoLineContainer
  moviesFound={2}
  page="details"
  sortBy="release_date"
>
  <div
    className="infoLine"
  >
    <DetailsResults>
      <div>
        Films by 
        Drama
         genre
      </div>
    </DetailsResults>
  </div>
</InfoLineContainer>
`;

exports[`InfoLineContainer Snapshot matches the snapshot for Home page: Home page 1`] = `
<InfoLineContainer
  moviesFound={2}
  page="home"
  sortBy="release_date"
>
  <div
    className="infoLine"
  >
    <DetailsResults>
      <div>
        Films by 
        Drama
         genre
      </div>
    </DetailsResults>
  </div>
</InfoLineContainer>
`;
